security/security-core/src/test/java/io/camunda/security/auth/OidcGroupsLoaderTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.security.auth;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+final class OidcGroupsLoaderTest {
+
+  @Test
+  void shouldIgnoreMissingConfig() {
+    // given
+    final var claims =
+        Map.<String, Object>of(
+            "groups", "testuser",
+            "other_claim", "other_value");
+
+    final var loader = new OidcGroupsLoader(null);
+
+    // when
+    final var groups = loader.load(claims);
+
+    // then
+    assertThat(groups).isNull();
+  }
+
+  @Test
+  void shouldSanitizeInvalidRegex() {
+    final var loader = new OidcGroupsLoader(".gg.ee...");
+    assertThat(loader.getGroupsClaim()).isEqualTo("$['.gg.ee...']");
+  }
+
+  @Test
+  void shouldThrowOnUnexpectedClaimType() {
+    // given
+    final var claims = Map.<String, Object>of("group", Map.of("names", List.of("g1", "g2")));
+    final var loader = new OidcGroupsLoader("$.group");
+    // when
+    assertThatThrownBy(() -> loader.load(claims))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Group's list derived from ($.group) is not a string array. Please check your OIDC configuration.");
+  }
+
+  @Test
+  void shouldLoadGroupsWithSimpleExpression() {
```

Comment:
🔧 Can you add a test here that uses a simple expression claim with special characters i.e. `http://test.com` as we had this issue before and it would be good to capture the same checks here in these tests so if we change behaviour elsewhere it it captured